src/pybind/CMakeLists.txt

```diff
@@ -67,6 +67,23 @@ if(WITH_MGR)
   elseif(NOT WITH_PYTHON3 AND MGR_PYTHON_VERSION_MAJOR EQUAL 3)
     message(FATAL_ERROR "mgr plugins require python3 binding")
   endif()
+
+  if (USE_OPENSSL)
+    execute_process(
+      COMMAND ${PYTHON${PYTHON_VERSION}_EXECUTABLE} -c "import ssl; print('.'.join(map(str,ssl.OPENSSL_VERSION_INFO[0:3])))"
```

Comment:
so, we only have this issue when running ceph-mgr with a python runtime linked against a different OpenSSL?

if that's the case, we might want to put `MGR_PYTHON_VERSION` here instead if `PYTHON_VERSION` . `PYTHON_VERSION` is for the supported version of python runtime of its binding.

also, might want to pass `OUTPUT_STRIP_TRAILING_WHITESPACE` to `execute_process()`.

Reply:
if it's required that we should use the same version of OpenSSL, i'd suggest put something like
```cmake
if(WITH_MGR)
  execute_process(
    COMMAND ${PYTHON${MGR_PYTHON_VERSION}_EXECUTABLE} -c "import ssl; 
  print('.'.join(map(str,ssl.OPENSSL_VERSION_INFO[0:3])))"
    RESULT_VARIABLE PYSSL_RESULT
    OUTPUT_VARIABLE PYSSL_VER
    ERROR_QUIET
    OUTPUT_STRIP_TRAILING_WHITESPACE)
  if(PYSSL_RESULT)
    message(FATAL "could not determine the OpenSSL version used by python ssl module")
  endif()
  find_package(OpenSSL ${PYSSL_VER} REQUIRED)
else(WITH_MGR)
  find_package(OpenSSL REQUIRED)
endif(WITH_MGR)
```
please note, i incorporated the suggested changes in the code snippet above.

Reply:
thanks, this looks good, I'll incorporate these. afaik the issue happens when multiple openssl libraries (maybe for old 1.0.x versions it may not be that prominent compared to 1.0.x vs 1.1.x changes) are loaded which can happen if system python ssl lib or any of the python c-modules are linked against a different openssl.

Reply:
Hmm, thinking a bit more into this; I'm not sure whether we need to ensure that we use system python's openssl version. In my own case my system python ssl was using 1.0.2; but python envs can be easily loaded using something like pyenv or so, my initial idea was just to detect this incompatibility  rather than force ceph to be linked against the openssl python ssl was linked against. Since this isn't that common a scenario in build machines, I'd love to hear your thoughts on this.

Reply:
> python envs can be easily loaded using something like pyenv or so

my question still persists: "so, we only have this issue when running ceph-mgr with a python runtime linked against a different OpenSSL?"

if it's mgr's python runtime, it's not simple to instruct mgr to use a different python runtime at runtime.

Reply:
> my question still persists: "so, we only have this issue when running ceph-mgr with a python runtime linked against a different OpenSSL?"

Yes we only have the issue when ceph-mgr is linked with a runtime different from python's openssl linkage. 

> if it's mgr's python runtime, it's not simple to instruct mgr to use a different python runtime at runtime.